src/AgentContainer.cpp

```diff
@@ -644,6 +666,49 @@ void AgentContainer::initAgentsCensus (iMultiFab& num_residents,    /*!< Number
                     school_ptr[ip] = -1;
                 }
 
+                // Increment the appropriate student counter based on the school assignment
+                if (school_ptr[ip] == 3) {
+                    amrex::Gpu::Atomic::AddNoRet(&student_counts_arr(i, j, 3), 1);  // Elementary school 3
+                } else if (school_ptr[ip] == 4) {
+                    amrex::Gpu::Atomic::AddNoRet(&student_counts_arr(i, j, 4), 1);  // Elementary school 4
+                } else if (school_ptr[ip] == 2) {
+                    amrex::Gpu::Atomic::AddNoRet(&student_counts_arr(i, j, 2), 1);  // Middle school
+                } else if (school_ptr[ip] == 1) {
+                    amrex::Gpu::Atomic::AddNoRet(&student_counts_arr(i, j, 1), 1);  // High school
+                } else if (school_ptr[ip] == 5) {
+                    amrex::Gpu::Atomic::AddNoRet(&student_counts_arr(i, j, 0), 1);  // Playgroups + Day care
+                }
+
+            }
+        });
+
+        /*
+        Code number
+        0 = daycare for now -- Handle Preschool later
+        1 = high school
+        2 = middle
+        3 = elementary neighborhood 1
+        4 = elementary neighborhood 2
+        */
+
+        amrex::ParallelFor(bx,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+        {
+            if (k == 0){ // avoid double compute
+                teacher_counts_arr(i, j, 0) = (student_counts_arr(i, j, 0) + 19)  / 20;
+                teacher_counts_arr(i, j, 1) = (student_counts_arr(i, j, 1) + 19)  / 20;
+                teacher_counts_arr(i, j, 2) = (student_counts_arr(i, j, 2) + 19)  / 20;
+                teacher_counts_arr(i, j, 3) = (student_counts_arr(i, j, 3) + 19)  / 20;
+                teacher_counts_arr(i, j, 4) = (student_counts_arr(i, j, 4) + 19)  / 20;
```

Comment:
Instead of hard-coding `20`, we can instead have an int vector of ratios. The default can be `[20 20 20 20 20]`, but can also be read in from input file?

Reply:
ratio can be read from input file now as agent.teacher_student_ratios